cluster-operator/src/main/java/io/strimzi/operator/cluster/model/JmxTrans.java

```diff
@@ -0,0 +1,364 @@
+ * JmxTrans deployment including: config map, deployment, and service accounts.
+ */
+public class JmxTrans extends AbstractModel {
+
+    // Configuration defaults
+    private static final int DEFAULT_HEALTHCHECK_DELAY = 15;
+    private static final int DEFAULT_HEALTHCHECK_TIMEOUT = 5;
+    private static final String DEFAULT_JMXTRANS_IMAGE = "strimzi/jmxtrans:latest";
+    public static final Probe LIVENESS_PROBE_OPTIONS = new ProbeBuilder().withTimeoutSeconds(DEFAULT_HEALTHCHECK_TIMEOUT).withInitialDelaySeconds(DEFAULT_HEALTHCHECK_DELAY).build();
+
+    // Configuration for mounting `config.json` to be used as Config during run time of the JmxTrans
+    public static final String JMXTRANS_CONFIGMAP_KEY = "config.json";
+    public static final String JMXTRANS_VOLUME_NAME = "jmx-config";
+    public static final String CONFIG_MAP_ANNOTATION_KEY = "config-map-revision";
+    protected static final String JMX_METRICS_CONFIG_SUFFIX = "-jmxtrans-config";
+    public static final String JMX_FILE_PATH = "/var/lib/jmxtrans";
+
+    private boolean isDeployed;
+    private boolean isJmxAuthenticated;
+    private String configMapName;
+    private String clusterName;
+
+    /**
+     * Constructor
+     *
+     * @param namespace Kubernetes/OpenShift namespace where JmxTrans resources are going to be created
+     * @param kafkaCluster kafkaCluster name
+     * @param labels    labels to add to the kafkaCluster
+     */
+    protected JmxTrans(String namespace, String kafkaCluster, Labels labels) {
+        super(namespace, kafkaCluster, labels);
+        this.name = JmxTransResources.deploymentName(kafkaCluster);
+        this.clusterName = kafkaCluster;
+        this.replicas = 1;
+        this.readinessPath = "/metrics";
+        this.livenessPath = "/metrics";
+        this.livenessProbeOptions = LIVENESS_PROBE_OPTIONS;
+
+        this.mountPath = "/var/lib/kafka";
+
+        // Metrics must be enabled as JmxTrans is all about gathering JMX metrics from the Kafka brokers and pushing it to remote sources.
+        this.isMetricsEnabled = true;
+    }
+
+    public static JmxTrans fromCrd(Kafka kafkaAssembly, KafkaVersion.Lookup versions) {
+        JmxTrans result = null;
+
+        KafkaJmxOptions spec = kafkaAssembly.getSpec().getKafka().getJmxOptions();
+        if (spec != null && spec.getJmxTransSpec() != null) {
+            result = new JmxTrans(kafkaAssembly.getMetadata().getNamespace(),
+                    kafkaAssembly.getMetadata().getName(),
+                    Labels.fromResource(kafkaAssembly).withKind(kafkaAssembly.getKind()));
+            result.isDeployed = true;
+
+            if (spec.getAuthentication() instanceof KafkaJmxAuthenticationPassword) {
+                result.isJmxAuthenticated = true;
+            }
+
+            result.setResources(spec.getJmxTransSpec().getResources());
+
+            result.setImage(DEFAULT_JMXTRANS_IMAGE);
```

Comment:
The image should be reconfigurable as for other deployments:
* Through the environment varialble in the CO deploment
* Through the `image` property in the  JmxTransSpec (if it is a separate image, through the `image` property in `spec.kafka` it is would be for example shared image with Kafka).

Reply:
Will add that in!